### ISSUE TITLE
cmd/compile: failure on a function type alias

### ISSUE THREAD

**Issue by falconandy** [golang/go#27232]
### What version of Go are you using (`go version`)?

go version go1.11 linux/amd64

### Does this issue reproduce with the latest release?

Yes

### What operating system and processor architecture are you using (`go env`)?

Without empty values:
GOARCH="amd64"
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/falconandy/go"
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build437619988=/tmp/go-build -gno-record-gcc-switches

### What did you do?

https://play.golang.org/p/ydDs-6YdK64

### What did you expect to see?

Compiled successfully

### What did you see instead?

cannot use parser literal (type *parser) as type Parser in return argument:
*parser does not implement Parser (wrong type for AddOperator method)
  have AddOperator(string, func(Parser))
   want AddOperator(string, Parser)

**Comment by cznic** [comment]
I'm probably missing something but the code linked at the Playground compiles and executes just fine. Can you please provide a reproducer that actually fails with the error in the OP?

**Comment by falconandy** [comment]
Playground works on Go 1.10.3 - so it's fine there

**Comment by cznic** [comment]
Thanks, I can reproduce the fail now. Seems like a genuine regression to me.

**Comment by agnivade** [comment]
@josharian @mdempsky 

Seems like a backport candidate.

**Comment by FiloSottile** [comment]
@gopherbot please file this for backport against 1.11. This is a regression.

@griesemer please [make a cherry-pick CL](https://github.com/golang/go/wiki/MinorReleases#making-cherry-pick-cls) once your change is merged in master.

**Comment by agnivade** [comment]
Roger.

**Comment by griesemer** [comment]
Simpler reproducer, with irrelevant details stripped:
```Go
package p

type F = func(T) // compiles if not type alias or moved below T interface

type T interface {
	m(F)
}

type t struct{}

func (t) m(F) {}

var _ T = &t{}
```

**Comment by griesemer** [comment]
Probably a duplicate of #25838. Leaving open for now.
